### PR TITLE
rEFInd 0.8.4 (new formula)

### PR DESCRIPTION
**Pull request by saulshanabrook**
A boot manager.

This is the updated forked of rEFIt, which was has stopped being
updated (#24818).

I couldn't figure out any easy way of testing the installation. 

Also uninstall [is possible but looks messy](http://www.rodsbooks.com/refind/installing.html#uninstalling) on OS X, so I haven't included that either.

### CODE REVIEW

File: Library/Formula/refind.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Refind < Formula
+  homepage "http://www.rodsbooks.com/refind/"
+  url "https://downloads.sourceforge.net/project/refind/0.8.4/refind-bin-0.8.4.zip"
+  sha1 "f219383924ef60cb6d147267f9ee44810c1f473f"
+
+  def install
+    system "sudo", "bash", "install.sh"
```
Comment (adamv) [inline Library/Formula/refind.rb]: We definitely can't allow a formula to run `sudo`; is a non-sudo install possible?

File: Library/Formula/refind.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  end
+end
```
Comment (mikemcquaid) [inline Library/Formula/refind.rb]: Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/refind.rb (hunk 1 above)
Comment (saulshanabrook) [inline Library/Formula/refind.rb]: Nope. It must be run with sudo, because it is installing a new boot loader for your mac.

**Comment by mikemcquaid** [comment]
Passing on this if it needs `sudo`.

File: Library/Formula/refind.rb (hunk 1 above)
Comment (saulshanabrook) [inline Library/Formula/refind.rb]: Does that mean homebrew can't include this package?

**Comment by saulshanabrook** [comment]
@mikemcquaid Is there a better place to put this formula, like an alternative repo? Refind seems to be a relatively common tool, used if you want to dual boot from a USB, and would be useful to be able to install through homebrew.

File: Library/Formula/refind.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Formula/refind.rb]: No, unfortunately.

**Comment by winmillwill** [comment]
The only problem here is that we're invoking install.sh in the formula. We could just download the tools and be done, which would let people manage versions of the tool with homebrew. Whether they choose to invoke it or not is their business, just like whether I invoke htop as root or not is my business.

**Comment by mikemcquaid** [comment]
@saulshanabrook Sorry, missed your message earlier. I guess you could run your own tap but generally anything that uses `sudo` is a bad fit for Homebrew and a better fit for a `.pkg` installer or similar.
@winmillwill I don't really see the advantage in just having Homebrew download the tools and require the user to run it manually to do anything.

**Comment by winmillwill** [comment]
@mikemcquaid We have a tool that allows you to install a new boot loader. Installing the tool and using the tool to install the boot loader are not the same thing.

I agree that ideally we would have a .pkg that handled all of the boot loader install and uninstall, but since such a package doesn't exist, it seems easier to at least let the users of the package manager download the package that does exist.

**Comment by mikemcquaid** [comment]
@winmillwill You can create your own tap which will allow you to do that.

**Comment by beeka** [comment]
FYI - I found this pull request by trying to use brew to install rEFInd... I was't wanting to install the tool on the mac but to aid setting up a bootable SD card. I would have been a bit surprised if it had replaced the boot loader on my active partition.

It would be handy to have brew to manage the location / versions of refind. It makes sense to me that installing and running the tool are separate activities (installing ffmpeg doesn't run it). Some thought might need putting into the package interface - probably wouldn't want install.sh in /usr/local/bin... perhaps a link from /usr/local/bin/refind-install.sh would be useful.

Steve.

**Pull request closed without merging**